kratos/sources/model_part_io.cpp

```diff
@@ -1774,6 +1775,60 @@ namespace Kratos
 
         KRATOS_CATCH("")
     }
+    template<class TDataType>
+    void ModelPartIO::WriteDataBlock(const TDataType& rThisData, const std::string& DataName){
+        typedef VariableComponent<VectorComponentAdaptor<array_1d<double, 3> > > array_1d_component_type;
+        std::unordered_set<std::string> variables;
+        
+        for(auto& data :rThisData){
+            for(auto& var:data.GetData()){
+                auto const& is_included = variables.find(var.first->Name());
+                if(is_included == variables.end()){
+                    variables.insert(var.first->Name());
```

Comment:
why do you need `variables`?
It is a local variable but I don't get what it is being used for